### PR TITLE
feat: [progressbar/update]: Updated component

### PR DESCRIPTION
**Pull request by goyalsonali**
You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- StyleLint: `retrigger stylelint`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- ImageSnapshot `retrigger ImageSnapshot`

### CODE REVIEW

**Review by satyam-harness — APPROVED** [review]

File: packages/uicore/src/components/MultiStepProgressIndicator/MultiStepProgressIndicator.css [hunk 1]
```diff
@@ -58,3 +58,17 @@
 .spacing {
   padding: 2px !important;
 }
+.dotNameSuccess {
+  margin-top: -26px;
```
Comment (vardanbansal-harness) [inline packages/uicore/src/components/MultiStepProgressIndicator/MultiStepProgressIndicator.css:62]: Can we not use negative values here? Negative value for `margin-top` could be replaced with a `margin-bottom` instead.

File: packages/uicore/src/components/MultiStepProgressIndicator/MultiStepProgressIndicator.css [hunk 2]
```diff
@@ -58,3 +58,17 @@
...
+  position: absolute;
+}
+
+.dotNameFailed {
+  color: var(--grey-500);
+  margin-top: -26px;
```
Comment (vardanbansal-harness) [inline packages/uicore/src/components/MultiStepProgressIndicator/MultiStepProgressIndicator.css:68]: Same as above.

File: packages/uicore/src/components/MultiStepProgressIndicator/MultiStepProgressIndicator.tsx [hunk 3]
```diff
@@ -9,23 +9,43 @@ import React from 'react'
...
+import { Text } from '../..'
 
 import css from './MultiStepProgressIndicator.css'
+import { Color } from '@harness/design-system'
 
 type StepStatus = 'TODO' | 'INPROGRESS' | 'FAILED' | 'SUCCESS'
 
 export interface MultiStepProgressIndicatorProps {
-  progressMap: Map<number, StepStatus>
+  progressMap: Map<number, { StepStatus: StepStatus; StepName?: string }>
```
Comment (vardanbansal-harness) [inline packages/uicore/src/components/MultiStepProgressIndicator/MultiStepProgressIndicator.tsx:20]: usually variable names such as `StepStatus`, except enums should start with a lower-case character.

**Review by vardanbansal-harness — CHANGES_REQUESTED** [review]
Can we try not using negative values in css?

**Review by vardanbansal-harness — APPROVED** [review]

**Pull request merged**